app/src/main/java/com/google/android/samples/socialite/ui/camera/CameraViewModel.kt

```diff
@@ -108,20 +131,64 @@ class CameraViewModel @Inject constructor(
         .setQualitySelector(QualitySelector.from(Quality.HIGHEST))
         .build()
 
+    private val videoCaptureBuilder = VideoCapture.Builder(recorder)
+    private lateinit var videoCaptureUseCase: VideoCapture<Recorder>
+
+    private val imageAnalysisUseCase = ImageAnalysis.Builder()
+        .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+        .build()
+
+    lateinit var greenScreenEffect: OverlayEffect
+    val backgroundRemovalThreshold = 0.8
+
+    lateinit var mask: Bitmap
+    lateinit var bitmap: Bitmap
+    var isGreenScreenProcessing = false
+
     private var currentRecording: Recording? = null
     private lateinit var recordingState: VideoRecordEvent
 
     init {
         val videoCaptureBuilder = VideoCapture.Builder(recorder)
         viewModelScope.launch {
-            val hdrCameraInfo = getHdrCameraInfo()
+            greenScreenEffect = OverlayEffect(
+                PREVIEW or IMAGE_CAPTURE or VIDEO_CAPTURE,
+                5,
+                Handler(Looper.getMainLooper()),
+                {},
+            )
 
-            if (hdrCameraInfo != null) {
-                Log.i(TAG, "Capturing HDR video")
-                videoCaptureBuilder.setDynamicRange(hdrCameraInfo)
-            }
+            imageAnalysisUseCase.setAnalyzer(
+                ContextCompat.getMainExecutor(application),
+                SelfieSegmentationAnalyzer(),
+            )
 
-            videoCaptureUseCase = videoCaptureBuilder.build()
+            greenScreenEffect.setOnDrawListener { frame ->
+                if (!::mask.isInitialized || !::bitmap.isInitialized) {
+                    // Do not change the drawing if the frame doesn’t match the analysis
+                    // result.
+                    return@setOnDrawListener true
+                }
+
+                // Clear the previously drawn frame.
+                frame.overlayCanvas.drawColor(Color.TRANSPARENT, PorterDuff.Mode.CLEAR)
+
+                // Create a Paint object to draw the mask layer.
+                val paint = Paint()
+                paint.xfermode = PorterDuffXfermode(PorterDuff.Mode.DST_IN)
```

Comment:
This can be allocated outside of the `setOnDrawListener` so you aren't allocating a new one on each frame.